editoast/src/core/mod.rs

```diff
@@ -233,83 +208,80 @@ pub struct Bytes;
-            CoreError::CoreResponseFormatError {
-                msg: err.to_string(),
-            }
-            .into()
+    fn from_bytes(bytes: &[u8]) -> Result<Self::Response, Error> {
+        serde_json::from_slice(bytes).map_err(|err| Error::CoreResponseFormatError {
+            msg: err.to_string(),
         })
     }
 }
 
 impl CoreResponse for Bytes {
     type Response = Vec<u8>;
 
-    fn from_bytes(bytes: &[u8]) -> Result<Self::Response> {
+    fn from_bytes(bytes: &[u8]) -> Result<Self::Response, Error> {
         Ok(Vec::from_iter(bytes.iter().cloned()))
     }
 }
 
 impl CoreResponse for () {
     type Response = ();
 
-    fn from_bytes(_: &[u8]) -> Result<Self::Response> {
+    fn from_bytes(_: &[u8]) -> Result<Self::Response, Error> {
         Ok(())
     }
 }
 
 #[allow(clippy::enum_variant_names)]
-#[derive(Debug, Error, EditoastError)]
-#[editoast_error(base_id = "coreclient")]
-pub enum CoreError {
+#[derive(Debug, Error, Derivative)]
+#[derivative(PartialEq)]
+pub enum Error {
     #[error("Cannot parse Core response: {msg}")]
-    #[editoast_error(status = 500)]
     CoreResponseFormatError { msg: String },
-    /// A fallback error variant for when no meaningful error could be parsed
-    /// from core's output.
-    #[error("Core error {}: {raw_error}", status.unwrap_or(500))]
-    #[editoast_error(status = status.unwrap_or(500))]
-    GenericCoreError {
-        status: Option<u16>,
-        url: String,
-        raw_error: String,
-    },
     #[error("Core returned an error in an unknown format")]
     UnparsableErrorOutput,
 
     #[error("Core connection closed before message completed. Should retry.")]
-    #[editoast_error(status = 500)]
     ConnectionClosedBeforeMessageCompleted,
     #[error("Core connection reset by peer. Should retry.")]
-    #[editoast_error(status = 500)]
     ConnectionResetByPeer,
     #[error("Core connection broken. Should retry.")]
-    #[editoast_error(status = 500)]
     BrokenPipe,
 
     #[error(transparent)]
-    #[editoast_error(status = "500")]
-    MqClientError(MqClientError),
+    MqClientError(#[from] MqClientError),
+
+    #[error(transparent)]
+    StandardCoreError(#[from] StandardCoreError),
 
     #[cfg(test)]
     #[error("The mocked response had no body configured - check out StubResponseBuilder::body if this is unexpected")]
     NoResponseContent,
 }
 
-#[derive(Debug, Deserialize)]
+impl Error {
+    fn parse(bytes: &[u8], url: String) -> Error {
+        // We try to deserialize the response as an StandardCoreError in order to retain the context of the core error
+        if let Ok(mut core_error) = <Json<StandardCoreError>>::from_bytes(bytes) {
+            core_error.context.insert("url".to_owned(), url.into());
+            return Error::StandardCoreError(core_error);
+        }
+        Error::UnparsableErrorOutput
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
 pub struct StandardCoreError {
     #[serde(skip)]
```

Comment:
😨

We should deserialize the status code as well (you can place the serde remote definition in `editoast_common`).

Reply:
As discussed, the refactoring required to handle "responses as errors" will be done after this PR is merged